Lib/test/test_time.py

```diff
@@ -489,6 +489,10 @@ def test_localtime_failure(self):
         self.assertRaises(OSError, time.localtime, invalid_time_t)
         self.assertRaises(OSError, time.ctime, invalid_time_t)
 
+        # Issue #26669: check for localtime() failure
+        self.assertRaises(ValueError, time.localtime, float("nan"))
+        self.assertRaises(ValueError, time.ctime, float("nan"))
```

Comment:
Ok for "functional" tests", but I also would like to see unit tests on the following _testcapi functions:

* pytime_object_to_time_t
* pytime_object_to_timeval
* pytime_object_to_timespec
* PyTime_FromSeconds
* PyTime_FromSecondsObject
* PyTime_AsSecondsDouble

See CPyTimeTestCase in test_time.py.